src/pages/pickups.tsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react'
+import { gql, useQuery, useMutation } from '@apollo/client'
+import { Box, Spinner, Heading, Image, Text, Flex, Button, Icon } from '@chakra-ui/core'
+import { Trackable, User, TrackableStatus } from '../generated/graphql'
+import { Link} from 'react-router-dom'
+import Header from '../components/header'
+import debug from 'debug'
+
+const log = debug("pages.pickups")
+
+const PICKUPS_PAGE_QUERY = gql`
+    query PickUpsPage($filters: GetTrackablesFilter) {
+        me {
+            did
+        }
+        getTrackables(filters: $filters) {
+            did
+            trackables {
+                did
+                status
+            }
+        }
+    }
+`
+const ACCEPT_JOB_MUTATION = gql`
+    mutation PickUpsPageAccept($input: AcceptJobInput!) {
+        acceptJob(input: $input) {
+            code
+        }
+    }
+`
+
+export function PickUpsPage() {
+    const { data, loading, error } = useQuery(PICKUPS_PAGE_QUERY)
+    if (loading) {
+        return (
+            <Box>
+                <Spinner />
+            </Box>
+        )
+    }
+
+    if (error) {
+        return (
+            <Box>
+                <Text>{error}</Text>
+            </Box>
+        )
+    }
+
+    log("pickups page data: ", data)
+
+    if (data.getTrackables?.trackables?.length == 0) {
+        return (<Box>
+            <Header />
+            <Flex mt={5} p={10} flexDirection="column">
+                <Box>
+                    <Box>
+                        <Heading>No donations available for pickup.</Heading>
+                    </Box>
+                </Box>
+            </Flex>
+        </Box>)
+    }
+
+    const available = data.getTrackables.trackables.filter((trackable: Trackable) => {
+        return trackable.status == TrackableStatus.Published
+    })
+
+    return (
+        <Box>
+            <Header />
+            <Flex mt={5} p={10} flexDirection="column">
+                <Box>
+                    <Box>
+                        <TrackableCollection trackables={available} user={data.me} />
+                    </Box>
+                </Box>
+            </Flex>
+        </Box>
+    )
+}
+
+function AcceptJobButton({trackable, user}:{trackable:Trackable, user:User}) {
+    const [acceptJob] = useMutation(ACCEPT_JOB_MUTATION)
+
+    const onClick = async ()=> {
+        await acceptJob({
+            variables: {
+                input: {
+                    trackable: trackable.did,
+                    user: user.did,
+                }
+            },
+        })
+        // FIXME: use react'y things
+        window.location.href = "/summary"
```

Comment:
I found `useHistory` to be refreshingly straightforward for this. You can import it from `react-router-dom` and then it's just `const history = useHistory()` and then `history.push("/summary")`. I'm using it in the donation flow here: https://github.com/quorumcontrol/simple-tracker/blob/2af88ae80ea4a4f1b32b61b99ec5081a9dbce3e6/src/pages/donate.tsx#L100

Reply:
ah i had tried that, but I think I put the use history in the wrong spot, will look ty